File: api/src/main/java/com/alten/remotesync/application/project/record/request/ProjectFilterDTO.java
package com.alten.remotesync.application.project.record.request;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;

public record ProjectFilterDTO(

        String filter,

        String sort,

        Integer sortType,

        String value,

        @NotNull
        @Min(value = 0, message = "Page size must be at least 1")
        Integer pageNumber,
        @NotNull
        @Min(value = 0, message = "Page size must be at least 1")
        Integer pageSize
) {
}

File: app/src/app/dto/rc/project-filter.dto.ts
/**
 * ProjectFilter interface that maps to the backend ProjectFilterDTO
 * @see Backend: com.alten.remotesync.application.project.record.request.ProjectFilterDTO
 */
export interface ProjectFilter {
    filter?: 'projectLabel' | 'clientLabel' | "",
    value?: string,
    sort: 'label' | 'titre' | 'client' | 'status'
    sortType: 1 | -1,
    pageNumber: number,
    pageSize: number,

}